mono/metadata/object.c

```diff
@@ -7120,68 +7113,18 @@ mono_string_handle_to_utf8 (MonoStringHandle s, MonoError *error)
 	return mono_string_to_utf8_checked (MONO_HANDLE_RAW (s), error);
 }
 
-/**
- * mono_string_to_utf8_ignore:
- * \param s a MonoString
- * Converts a \c MonoString to its UTF-8 representation. Will ignore
- * invalid surrogate pairs.
- * The resulting buffer should be freed with \c mono_free().
- */
-char *
-mono_string_to_utf8_ignore (MonoString *s)
-{
-	MONO_REQ_GC_UNSAFE_MODE;
-
-	long written = 0;
-	char *as;
-
-	if (s == NULL)
-		return NULL;
-
-	if (!s->length)
-		return g_strdup ("");
-
-	as = g_utf16_to_utf8 (mono_string_chars (s), s->length, NULL, &written, NULL);
-
-	/* g_utf16_to_utf8  may not be able to complete the convertion (e.g. NULL values were found, #335488) */
-	if (s->length > written) {
-		/* allocate the total length and copy the part of the string that has been converted */
-		char *as2 = (char *)g_malloc0 (s->length);
-		memcpy (as2, as, written);
-		g_free (as);
-		as = as2;
-	}
-
-	return as;
-}
-
 /**
  * mono_string_to_utf8_image_ignore:
  * \param s a \c System.String
- * Same as \c mono_string_to_utf8_ignore, but allocate the string from the image mempool.
+ * Same as \c mono_string_to_utf8_checked, but ignore error
+ * and allocate the string from the image mempool.
  */
 char *
 mono_string_to_utf8_image_ignore (MonoImage *image, MonoString *s)
 {
-	MONO_REQ_GC_UNSAFE_MODE;
-
-	return mono_string_to_utf8_internal (NULL, image, s, TRUE, NULL);
+	return mono_string_to_utf8_internal (NULL, image, s, NULL);
 }
 
-/**
- * mono_string_to_utf8_mp_ignore:
- * \param s a \c System.String
- * Same as \c mono_string_to_utf8_ignore, but allocate the string from a mempool.
- */
-char *
-mono_string_to_utf8_mp_ignore (MonoMemPool *mp, MonoString *s)
-{
-	MONO_REQ_GC_UNSAFE_MODE;
-
-	return mono_string_to_utf8_internal (mp, NULL, s, TRUE, NULL);
```

Comment:
Now that `mono_string_to_utf8_mp_ignore`, if we only updated the single caller of `mono_string_to_utf8_image_ignore` (in `sre.c`) to _not_ ignore the error, we could be rid of all the error-ignoring cases of `mono_string_to_utf8_internal`, I think.